### PR TITLE
REF: Apply.apply_multiple

### PR DESCRIPTION
**Pull request by topper-123**
A step towards fixing https://github.com/pandas-dev/pandas/pull/53325#discussion_r1201397825.

This avoids the need to call out into `(Series|DataFrame).aggregate`. In a follow-up I will replace the use of `agg_(list|dict)_like` with `apply` specific versions of those methds or some similar solution, avoiding needing to call `Apply.agg` from within `Apply.apply_multiple`.

CC @rhshadrach

### CODE REVIEW

File: pandas/core/apply.py [hunk 1]
```diff
@@ -554,7 +561,20 @@ def apply_multiple(self) -> DataFrame | Series:
         result: Series, DataFrame, or None
             Result when self.f is a list-like or dict-like, None otherwise.
         """
-        return self.obj.aggregate(self.f, self.axis, *self.args, **self.kwargs)
+        if self.axis == 1 and isinstance(self.obj, ABCDataFrame):
+            return self.obj.T.apply(self.f, 0, *self.args, **self.kwargs).T
```
Comment (rhshadrach) [inline pandas/core/apply.py:565]: Needs to be `args=self.args`, not `*self.args`. Can you add a test here?

File: pandas/core/apply.py [hunk 2]
```diff
@@ -135,13 +141,14 @@ def __init__(
         ):
 
             def f(x):
+                assert callable(func)
                 return func(x, *args, **kwargs)
```
Comment (rhshadrach) [inline pandas/core/apply.py:145]: Just an aside: we do this in several places; I've never understood what the benefit of currying is. I think things would be simpler if we just removed it (we wouldn't need orig_f).

**Review by rhshadrach — CHANGES_REQUESTED** [review]

File: pandas/core/apply.py (hunk 2 above)
Comment (topper-123) [reply pandas/core/apply.py:145]: Yeah, agree.

File: pandas/core/apply.py (hunk 1 above)
Comment (topper-123) [reply pandas/core/apply.py]: Ye, good catch. Done.

**Review by rhshadrach — APPROVED** [review]
lgtm

**Pull request merged**